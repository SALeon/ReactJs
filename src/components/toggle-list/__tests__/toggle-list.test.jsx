import React from 'react';
import { shallow, mount } from 'enzyme';
import ToggleList from '../index';

const mockProps = [
  {
    id: '1',
    component: (<div>mock</div>),
  },
  {
    id: '2',
    component: (<div>mock</div>),
  },
  {
    id: '3',
    component: (<div>mock</div>),
  },
];

describe('ToggleList', () => {
  describe('with shallow', () => {
    let shallowWrapper;
    beforeEach(() => {
      shallowWrapper = shallow(<ToggleList items={mockProps} />);
    });

    it('set default selected item', () => {
      expect(shallowWrapper.state().condition).toBe(mockProps[0].id);
    });

    it('snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('with mount', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(<ToggleList items={mockProps} />);
    });

    it('toggle filter', () => {
      wrapper.find('.toggle-item').at(1).simulate('click');
      expect(wrapper.state().condition).toBe(mockProps[1].id);
    });

    it('not toggle filter', () => {
      const sourceState = `${wrapper.state().condition}`;
      wrapper.find('.toggle-item').at(0).simulate('click');
      expect(wrapper.state().condition).toBe(sourceState);
    });
  });
});
