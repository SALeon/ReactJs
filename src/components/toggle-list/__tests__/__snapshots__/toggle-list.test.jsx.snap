// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleList with shallow snapshot 1`] = `
<ul>
  <li
    key="1"
  >
    <ToggleItem
      component={
        <div>
          mock
        </div>
      }
      id="1"
      selectedId="1"
      toggleListener={[Function]}
    />
  </li>
  <li
    key="2"
  >
    <ToggleItem
      component={
        <div>
          mock
        </div>
      }
      id="2"
      selectedId="1"
      toggleListener={[Function]}
    />
  </li>
  <li
    key="3"
  >
    <ToggleItem
      component={
        <div>
          mock
        </div>
      }
      id="3"
      selectedId="1"
      toggleListener={[Function]}
    />
  </li>
</ul>
`;
